### DISCUSSION TITLE
Literal types with single value not displaying correctly in docs (swagger/redoc)

### DISCUSSION THREAD

**Discussion by jordantshaw** [fastapi/fastapi#10218] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typing import List, Literal
from fastapi import FastAPI, Query

app = FastAPI()


@app.get('/test')
def test(
        foo: List[Literal['value1']] = Query(None),  # Single value in Literal does not display correctly in docs
        bar: List[Literal['value1', 'value2']] = Query(None),  # Multiple values in Literal displays as expected

):
    return {}
```

### Description

When using the 'Literal' type in a query parameter I would expect any and all of the Literal options to be listed with the query parameter in the docs.  However it appears when only a single value is used in Literal (`Literal['foo']`), then the docs are not displaying correctly.  The query parameter is listed but does not show the single value as an option.

### Operating System

macOS

### FastAPI Version

0.103.0

### Pydantic Version

2.3.0

### Python Version

Python 3.10.12

**Comment by YuriiMotov** [comment]
It can be reproduced with even simpler example:
```python
from typing import Literal
from fastapi import FastAPI

app = FastAPI()

@app.get('/test2')
def test_2(foo: Literal['value1']):
    return {}
```
Open Swagger docs.
Expected: dropdown list with possible values
Actual: string field

In openapi schema this field is rendered as:
```json
        "parameters": [
          {
            "name": "foo",
            "in": "query",
            "required": true,
            "schema": {
              "const": "value1",
              "type": "string",
              "title": "Foo"
            }
          }
        ],
```
Changing `"const": "value1",` to `"enum": ["value1"],` in schema would solve the problem.

Everything works well with enums:
```python
from enum import Enum
from fastapi import FastAPI

app = FastAPI()

class MyV(str, Enum):
    value1 = "value1"

@app.get("/test")
def test(foo: MyV):
    return {}
```
